packages/ember-glimmer/lib/environment.js

```diff
@@ -72,7 +73,13 @@ export default class Environment extends GlimmerEnvironment {
       }
     }
 
-    return super.refineStatement(statement);
+    let nativeSyntax = super.refineStatement(statement);
+
+    if (!nativeSyntax && key && this.hasHelper(key)) {
```

Comment:
So I think we want to actually pass all of this as the second argument to the `assert` if possible. The way that assert works is that it is only there in dev build. The production build strips the `assert`, so in the prod build we actually would have something like this:

```
if (!nativeSyntax && key && this.hasHelper(key)) {}
```

You may want to do an IIFE in the assertion to make it a little bit easer to read.